Modules/posixmodule.c

```diff
@@ -1888,7 +1888,17 @@ win32_xstat_impl(const wchar_t *path, struct _Py_stat_struct *result,
             /* Try reading the parent directory. */
             if (!attributes_from_dir(path, &fileInfo, &tagInfo.ReparseTag)) {
                 /* Cannot read the parent directory. */
-                SetLastError(error);
+                switch (GetLastError()) {
+                case ERROR_FILE_NOT_FOUND: /* File cannot be found */
+                case ERROR_PATH_NOT_FOUND: /* File parent directory cannot be found */
+                case ERROR_NOT_READY: /* Drive exists but unavailable */
+                case ERROR_BAD_NET_NAME: /* Remote drive unavailable */
+                    break;
+                /* Restore the error from CreateFileW(). */
+                default:
+                    SetLastError(error);
+                }
+
                 return -1;
```

Comment:
By the way, why is this done here instead of:

```python
retval = -1;
goto cleanup;
```

as done in other places in this function?

// cc @zooba

Reply:
From reading the code, I'd guess because there's nothing to clean up at this stage. Despite (now) being the majority of the code, this is still initialisation ;-)

Reply:
Makes sense, however `goto cleanup` is also done here, for example:

https://github.com/python/cpython/blob/8e3fde728f547f1d32bde8adf62b4c50bb877b9d/Modules/posixmodule.c#L1934-L1940

However I didn't notice anything being initialized after the previous error handling, where a simple `return -1` was done.

Reply:
Cleanup is required in that case to close `hFile`.

Reply:
Ah, got it. I didn't notice the error handling prior to the code snippet above was part of a big `switch` statement, which handled errors in the initialization of `hFile` :)